runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/StreamingEngineClient.java

```diff
@@ -102,7 +105,8 @@ private StreamingEngineClient(
       WindmillStubFactory stubFactory,
       GetWorkBudgetDistributor getWorkBudgetDistributor,
       GrpcDispatcherClient dispatcherClient,
-      long clientId) {
+      long clientId,
+      ChannelCache channelCache) {
```

Comment:
It seems like the channelcache is overlapping with the stubfactory.
Maybe it would be better to merge them by taking in a ChannelCachingRemoteStubFactory?

Reply:
done
Added another interface that extends WindmillStubFactory to expose some of the caching behavior.